app/UI/app.tsx

```diff
@@ -396,8 +395,8 @@ export class AppComponent extends Component<AppProps, {
                             newMessageChecker: app.conversationLists,
                             relayRecordGetter: app.database,
                             pinListGetter: app.otherConfig,
-                            getProfileByPublicKey: app.database.getProfileByPublicKey,
-                            getProfilesByText: app.database.getProfilesByText,
+                            getProfileByPublicKey: app.database.getProfileByPublicKey(model.currentRelay),
```

Comment:
This is not a good way to currying.

Currying intorudce hidden state, possibly mutable. We only do it when that hidden state is irrelevant to the downstream or difficult to retrieve for the downstream.

Therefore, most of the time, currying is used for dependency injections.